app/scripts/controllers/swaps.types.ts

```diff
@@ -121,6 +323,45 @@ export type QuoteRequest = {
   walletAddress: string;
 };
 
+export type SwapsControllerOptions = {
+  getBufferedGasLimit: (
+    params: {
+      txParams: {
+        value: string;
+        data: string;
+        to: string;
+        from: string;
+      };
+    },
+    factor: number,
+  ) => Promise<{ gasLimit: string; simulationFails: boolean }>;
+  provider: ExternalProvider | JsonRpcFetchFunc;
+  getProviderConfig: () => ProviderConfig;
+  getTokenRatesState: () => {
+    marketData: Record<
+      string,
+      {
+        [tokenAddress: string]: {
+          price: number;
+        };
+      }
+    >;
+  };
+  fetchTradesInfo: typeof defaultFetchTradesInfo;
+  getCurrentChainId: () => ChainId;
```

Comment:
Same as above. We should replace this with messenger action(s).

Reply:
Updated!